c10/core/dispatch/OpSchema.h

```diff
@@ -32,16 +29,17 @@ inline DeviceTypeId to_device_type_id(DeviceType device_type) {
   }
 }
 
-// TODO get rid of tensor_to_dispatch_key once c2::Tensor is de-templatized. This then fits into a template lambda instead of a functor.
-struct tensor_to_dispatch_key final {
-    template<class TensorType>
-    TensorParameterDispatchKey operator()(const TensorType& tensor) const {
-      return TensorParameterDispatchKey{
-          to_device_type_id(tensor.GetDeviceType()),
-          LayoutId(0),
-          tensor.dtype().id()};
-    }
-};
+inline TensorParameterDispatchKey tensor_to_dispatch_key(const C10Tensor& tensor) {
+  return TensorParameterDispatchKey{
+      to_device_type_id(tensor.impl()->device_type()),
+      LayoutId(0),
+      tensor.impl()->dtype().id()};
+}
+
+template<size_t num_dispatch_args, size_t num_tensor_args, size_t... indices>
```

Comment:
What does this function do? Docs

Reply:
I don't think we need to document private functions. I added a short comment though.

Reply:
The comment is added in https://github.com/pytorch/pytorch/pull/15407/files